cypress/page-objects/signingPage.js

```diff
@@ -0,0 +1,115 @@
+import BasePage from './basePage';
+
+class SigningPage extends BasePage {
+  get signInEmailField() {
+    return cy.get('input[data-test="sign-in-email"]');
+  }
+
+  get signInPasswordField() {
+    return cy.get('input[data-test="sign-in-password"]');
+  }
+
+  get signInCTA() {
+    return cy.get('button').contains('Sign In');
+  }
+
+  get signInError() {
+    return cy.get('[data-test="signing-error"]');
+  }
+
+  get emptySignInEmailField() {
+    return cy.get('input[data-test="sign-in-email"]:invalid');
+  }
+
+  get emptySignInPasswordField() {
+    return cy.get('input[data-test="sign-in-password"]:invalid');
+  }
+
+  get signUpNameField() {
+    return cy.get('input[data-test="sign-up-name"]');
+  }
+
+  get signUpEmailField() {
+    return cy.get('input[data-test="sign-up-email"]');
+  }
+
+  get signUpPasswordField() {
+    return cy.get('input[data-test="sign-up-password"]');
+  }
+
+  get signUpConfirmPasswordField() {
+    return cy.get('input[data-test="sign-up-confirm-password"]');
+  }
+
+  get signUpCTA() {
+    return cy.get('button[data-test="sign-up-submit"]');
+  }
+
+  get signUpError() {
+    return cy.get('div[data-test="signing-error"]');
+  }
+
+  get emptySignUpNameField() {
+    return cy.get('input[data-test="sign-up-name"]:invalid');
+  }
+
+  get emptyOrInvalidSignUpEmailField() {
+    return cy.get('input[data-test="sign-up-email"]:invalid');
+  }
+
+  get emptyPasswordField() {
+    return cy.get('input[data-test="sign-up-password"]:invalid');
+  }
+
+  get emptyConfirmPasswordField() {
+    return cy.get('input[data-test="sign-up-confirm-password"]:invalid');
+  }
+
+  enterSignInEmail(email) {
+    this.signInEmailField.type(email);
+  }
+
+  enterSignInPassword(password) {
+    this.signInPasswordField.type(password);
+  }
+
+  submitSignInForm() {
+    this.signInCTA.click();
+  }
+
+  signIn(email, password) {
+    this.enterSignInEmail(email);
+    this.enterSignInPassword(password);
+    this.submitSignInForm();
+  }
+
+  enterSignUpName(name) {
+    this.signUpNameField.type(name);
+  }
+
+  enterSignUpEmail(email) {
+    this.signUpEmailField.type(email);
+  }
+
+  enterSignUpPassword(password) {
+    this.signUpPasswordField.type(password);
+  }
+
+  enterSignUpConfirmPassword(confirmPassword) {
+    this.signUpConfirmPasswordField.type(confirmPassword);
+  }
+
+  submitSignUpForm() {
+    this.signUpCTA.click();
+  }
+
+  signUp(name, email, password, confirmPassword) {
+    this.enterSignUpName(name);
+    this.enterSignUpEmail(email);
+    this.enterSignUpPassword(password);
+    this.enterSignUpConfirmPassword(confirmPassword);
+    this.submitSignUpForm();
+  }
+}
```

Comment:
```suggestion
  signUp(name, email, password, confirmPassword) {
    if (name) {
      this.enterSignUpName(name);
    }
    if (email) {
      this.enterSignUpName(name);
    }
    if (password) {
      this.enterSignUpPassword(password);
    }
    if (confirmPassword) {
      this.enterSignUpConfirmPassword(confirmPassword);
    }
    this.submitSignUpForm();
  }
}
```

Take a look at this and think how you can reduce amount of lines for your tests while using conditionals and especially for tests with missing data